next-app/src/lib/forms/Input.svelte

```diff
@@ -0,0 +1,36 @@
+<script>
+	import { onMount } from 'svelte'
+
+	export let label = ''
+	export let type = 'text'
+	export let value = ''
+	export let required = false
+	export let autofocus = false
+
+	let id = randomId()
+	let input = {}
+
+	onMount(autofocusIfRequested)
+
+	function randomId() {
+		return Math.random().toString(36).substring(2, 7)
+	}
+
+	function autofocusIfRequested() {
+		autofocus && input.focus()
+	}
+
+	// works around "svelte(invalid-type)" warning, i.e., can't have a dynamic type AND bind:value...keep an eye on https://github.com/sveltejs/svelte/issues/3921
+	function typeWorkaround(node) {
```

Comment:
I see the benefit here for using `type=password` yet having just one Input component instead of two. I think for things like checkboxes and radio buttons we'll want components named Checkbox and Radio, and so on, rather than re-using the Input component.

The one thing I'd want to check is whether this can cause a flash of unstyled content, and more importantly, whether there are any browsers that won't take the `type = 'password'` assignment after DOM rendering, and would therefore cause this to act like a regular text input where the password is visible. I don't *think* there are any such browsers, but I'd like some certainty.

The alternative, where we create a duplicate component called PasswordInput that functions almost identically to an Input component, is code duplication, and I'm fine with this solution **if** no browsers exist where it could expose someone's password to a shoulder-surfer. But I don't know if all mobile browsers (mobile Opera, for instance) would do that.

Reply:
> I see the benefit here for using type=password yet having just one Input component instead of two. I think for things like checkboxes and radio buttons we'll want components named Checkbox and Radio, and so on, rather than re-using the Input component.

yeah, those might make sense to do for the same reasons we wouldn't include `input type=button` in here but there will be others like `email`, `search` that are similar enough not write a separate comp so it will boil down to a case-by-case basis.  I assume we'll at least do wrappers for the corresponding UI comps from Daisy:
![image](https://user-images.githubusercontent.com/4412848/161817291-f6d08a5f-8740-4e1c-a551-6325782c77f1.png)

> The one thing I'd want to check is whether this can cause a flash of unstyled content

If I'm thinking about the scenario correctly, this would only be a problem if we were to prefill the (password) input with a value which I don't think we ever would.  If there's another scenario I'm not thinking of let me know and we'll REPL it to make sure.

> whether there are any browsers that won't take the type = 'password' assignment after DOM rendering

I'm not familiar with this possibility but if it turns out to be a real concern we can look at mitigating it.

> The alternative, where we create a duplicate component...
yeah, just putting a conditional around the two concrete inputs would be another possible solution.  

We can look at solutions to mitigate your concerns if you find they are still valid.